docs/content/guides/developer/getting-started/data-serving.mdx

```diff
@@ -0,0 +1,115 @@
+
+GraphQL has been a key ask from many application builders, and the overall combination will come with following capabilities:
+
+- Indexer 2.0 will be a declarative framework that you would be able to configure to load different kinds of network data into postgres relational tables in parallel. This change is being implemented to improve the performance of the data ingestion into the postgres database. 
+  - In addition, you would be able to configure pruning for different tables in the postgres tables, allowing you to tune it for the desired combination of performance and cost characteristics.
+- GraphQL Server will be a lightweight server component that you would be able to use to access the data from different tables in the postgres database. This component will allow for combination of expressive queries due to the inherent nature of GraphQL, which is appealing to the frontend developers.
+
+:::info
+
+GraphQL with Indexer 2.0 is in beta and its recommended that you thoroughly investigate its functional, performance and cost properties before using it in production. Check out [this getting started document](guides/developer/getting-started/graphql-rpc.mdx) that refers to Mysten Labs managed relevant infrastructure.
+
+A sneak preview of how Indexer 2.0 could be setup is available [here](https://github.com/amnn/sui-sender-indexer).
+
+:::
+
+See [below](#when-to-use-grpc-vs-graphql-with-indexer-20) for a comparison with the gRPC API.
+
+### When to use gRPC vs GraphQL with Indexer 2.0
+
+You can use the high-level criteria mentioned in the following table to determine whether gRPC API or GraphQL RPC with Indexer 2.0 would better serve your use case. It's not an exhaustive list and it's expected that either of the options could work suitably for some of the use cases.
+
+| **Dimension** | **gRPC API** | **GraphQL RPC with Indexer 2.0** |
+| -------- | ------- | ------- |
+| Type of application or data consumer | Ideal for Web3 exchanges, defi market maker apps, other defi protocols or apps with ultra low-latency needs | Ideal for webapp builders or builders with slightly relaxed latency needs |
+| Query patterns | Okay to read data from different endpoints separately and combine on the client-side; faster serialization, parsing and validation due to binary format | Need easier docoupling of the client with the ability to combine different types of data in fewer calls |
+| Retention period requirements | Default retention period will be 2 weeks with actual configuration dependent on the Full node operator and their needs & goals; see history-related note after the table | Default retention period in postgres database will be 4 weeks with actual configuration depending on your or the data indexer operator's needs; see history-related note after the table |
```

Comment:
For GraphQL, the 4 week retention period applies to queries (like "give me all transactions that called these functions") and not to point look-ups, which will use the archival service and therefore are not pruned. This is maybe another useful differentiator between the services?

Reply:
I think we should wait to be more specific here until after we've figured out and have alignment on the historical data architecture. IMO, we would need some mechanism for exchanges or defi protocol builders to access historical data when using gRPC, so we need to think through this.